pkg/query-service/app/preferences/model.go

```diff
@@ -0,0 +1,598 @@
+	}
+
+	if len(allUserPreferencesWithGroups) == 0 {
+		return nil, nil
+	}
+
+	orgPreferenceValues := []PreferenceKV{}
+	query = `
+	SELECT preference_id, preference_value FROM org_preference WHERE org_id=$1;
+	`
+	err = db.Select(&orgPreferenceValues, query, orgId)
+
+	if err != nil {
+		return nil, &model.ApiError{Typ: model.ErrorExec, Err: fmt.Errorf("error in getting all org preference values: %s", err)}
+	}
+
+	orgPreferenceValueMap := map[string]string{}
+
+	for _, preferenceValue := range orgPreferenceValues {
+		orgPreferenceValueMap[preferenceValue.PreferenceId] = preferenceValue.PreferenceValue
+	}
+
+	userPreferenceValues := []PreferenceKV{}
+	query = `
+	SELECT preference_id, preference_value FROM user_preference WHERE user_id=$1;
+	`
+	err = db.Select(&userPreferenceValues, query, user.Id)
+
+	if err != nil {
+		return nil, &model.ApiError{Typ: model.ErrorExec, Err: fmt.Errorf("error in getting all user preference values: %s", err)}
+	}
+
+	userPreferenceValueMap := map[string]string{}
+
+	for _, preferenceValue := range userPreferenceValues {
+		userPreferenceValueMap[preferenceValue.PreferenceId] = preferenceValue.PreferenceValue
+	}
+
+	groupPreferenceMap := map[string][]Preferences{}
+
+	for _, preference := range allUserPreferencesWithGroups {
+		preferenceWithValue := Preferences{}
+		if preference.OrgScope == 1 {
+			preferenceWithValue.Value = orgPreferenceValueMap[preference.Id]
+		}
+
+		if preference.UserScope == 1 {
+			if value, seen := userPreferenceValueMap[preference.Id]; seen {
+				preferenceWithValue.Value = value
+			}
+		}
+		if preferenceWithValue.Value == "" {
+			preferenceWithValue.Value = preference.DefaultValue
+		}
+		preferenceWithValue.Id = preference.Id
+		preferenceWithValue.Name = preference.Name
+		preferenceWithValue.DefaultValue = preference.DefaultValue
+		preferenceWithValue.GroupId = preference.GroupId
+		preferenceWithValue.DependsOn = preference.DependsOn
+		preferenceWithValue.UserScope = preference.UserScope
+		preferenceWithValue.OrgScope = preference.OrgScope
+
+		value, seen := groupPreferenceMap[preference.GroupId]
+		if !seen {
+			groupPreferenceMap[preference.GroupId] = append([]Preferences{}, preferenceWithValue)
+		} else {
+			groupPreferenceMap[preference.GroupId] = append(value, preferenceWithValue)
+		}
+	}
+
+	preferenceGroups := []PreferenceGroup{}
+
+	query = `SELECT * FROM preference_group;`
+
+	err = db.Select(&preferenceGroups, query)
+
+	if err != nil {
+		return nil, &model.ApiError{Typ: model.ErrorExec, Err: fmt.Errorf("error in getting all preference groups: %s", err)}
+	}
+
+	allUserPreferenceTree := buildGroupTree(preferenceGroups, "", groupPreferenceMap)
+	fmt.Println(allUserPreferenceTree)
+
+	return &allUserPreferenceTree, nil
+}
+
+// recursively create the preference group tree
+func buildGroupTree(groups []PreferenceGroup, parentID string, groupPreferenceMap map[string][]Preferences) []AllPreferenceResponse {
```

Comment:
Yet to understand this.

Reply:
we can sync to understand this if required